### PR TITLE
Numerous minor edits for formatting, grammar, consistency

### PR DESCRIPTION
**Pull request by melophonic**
- Grammar/diction changes for clarity
- Syntax highlighting
   - Specify langs where missing
   - Remove unnecessary left indents
- Rework project structure tree
- Fix misc. markdown issues

### CODE REVIEW

File: _docs/02.5-deploy_app.markdown [hunk 1]
```diff
@@ -33,7 +33,7 @@ Password:
 
 You should see a terminal message `Logged in as your-example-email@walmartlabs.com`. Now, let's deploy!
 
-We need to specify the [version of node](https://devcenter.heroku.com/articles/node-best-practices) you are using on your machine into our `package.json`. Add the code below the `author` section and before the `contributors` section of your `package.json`:
+You'll need to specify the [version of node](https://devcenter.heroku.com/articles/node-best-practices) you are using on your machine into our `package.json`. Add the following json below the `author` section and before the `contributors` section in your `package.json`:
```
Comment (caoyangs) [inline _docs/02.5-deploy_app.markdown]: Add the following json below the `author` section and before the `contributors` section in your `package.json`

This is not a requirement, you can add it anywhere inside `package.json`

File: _docs/03.2-build_server_plugin.markdown [hunk 2]
```diff
@@ -15,97 +15,97 @@ We are going to use the [GitHub Api](https://developer.github.com/v3/) to grab t
...
-        |_index.html
+server
+  |_plugins
+    |_friends
       |_index.js
-    |_views
-      |_index-view.jsx
+    |_webapp
+      |_index.html
     |_index.js
```
Comment (caoyangs) [inline _docs/03.2-build_server_plugin.markdown]: this `index.js` should fall under `webapp`

**Comment by caoyangs** [comment]
2 comments, rest LGTM

File: _docs/03.2-build_server_plugin.markdown (hunk 2 above)
Comment (melophonic) [inline _docs/03.2-build_server_plugin.markdown]: @caoyangs is there not also one under `server` ([looking at this](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/server/index.js)). The full tree snippet:

```
server
    ├── index.js
    ├── plugins
    │   └── webapp
    │       ├── index.html
    │       └── index.js
    └── views
        └── index-view.jsx
```

Comment (caoyangs) [inline _docs/03.2-build_server_plugin.markdown]: this is correct.

**Pull request merged**